common/autoware_universe_utils/include/autoware/universe_utils/ros/static_transform_buffer.hpp

```diff
@@ -0,0 +1,141 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+#define AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+
+#include "autoware/universe_utils/ros/transform_listener.hpp"
+
+#include <eigen3/Eigen/Core>
+#include <pcl_ros/transforms.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace std
+{
+template <>
+struct hash<std::pair<std::string, std::string>>
+{
+  size_t operator()(const std::pair<std::string, std::string> & p) const
+  {
+    size_t h1 = std::hash<std::string>{}(p.first);
+    size_t h2 = std::hash<std::string>{}(p.second);
+    return h1 ^ (h2 << 1u);
+  }
+};
+}  // namespace std
+
+namespace autoware::universe_utils
+{
+using std::chrono_literals::operator""ms;
+using Key = std::pair<std::string, std::string>;
+struct PairEqual
+{
+  bool operator()(const Key & p1, const Key & p2) const
+  {
+    return p1.first == p2.first && p1.second == p2.second;
+  }
+};
+using TFMap = std::unordered_map<Key, Eigen::Matrix4f, std::hash<Key>, PairEqual>;
+
+class StaticTransformBuffer
+{
+public:
+  StaticTransformBuffer() = default;
+
+  bool getTransform(
+    rclcpp::Node * node, const std::string & target_frame, const std::string & source_frame,
+    Eigen::Matrix4f & eigen_transform)
+  {
+    auto key = std::make_pair(target_frame, source_frame);
+    auto key_inv = std::make_pair(source_frame, target_frame);
+
+    // Check if the transform is already in the buffer
+    auto it = buffer_.find(key);
+    if (it != buffer_.end()) {
+      eigen_transform = it->second;
+      return true;
+    }
+
+    // Check if the inverse transform is already in the buffer
+    auto it_inv = buffer_.find(key_inv);
+    if (it_inv != buffer_.end()) {
+      eigen_transform = it_inv->second.inverse();
+      buffer_[key] = eigen_transform;
+      return true;
+    }
+
+    // Check if transform is needed
+    if (target_frame == source_frame) {
+      eigen_transform = Eigen::Matrix4f::Identity();
+      buffer_[key] = eigen_transform;
+      return true;
+    }
+
+    // Get the transform from the TF tree
+    auto tf_listener = std::make_shared<autoware::universe_utils::TransformListener>(node);
+    auto tf = tf_listener->getTransform(
+      target_frame, source_frame, rclcpp::Time(0), rclcpp::Duration(1000ms));
+    RCLCPP_DEBUG(
+      node->get_logger(), "Trying to enqueue %s -> %s transform to static TFs buffer...",
+      target_frame.c_str(), source_frame.c_str());
+    if (tf == nullptr) {
+      eigen_transform = Eigen::Matrix4f::Identity();
```

Comment:
returning false and setting the return value `eigen_transform = Eigen::Matrix4f::Identity();` at the same time is a bit weird.  If you want to modify existing lvalue Matrix like a huge member variable, then it is OK.

Otherwise is it possible to return optionla<Matrix> instead ?

Reply:
Thank you @soblin for your review.
Using optional was my first thought, but from user point of view is more convenient to already have such a transformation in matrix. Less code need on user side and it is still safe since it returns success status.

Reply:
All right. I think it is confusing that returning `false` status while causing the mutable side effect `eigen_transform = Eigen::Matrix4f::Identity();` at the same time. Can you note that as a doxygen docstring ?

Reply:
Good idead! Addressed in https://github.com/autowarefoundation/autoware.universe/pull/8174/commits/11c61beebc03a8671aedd16e3e069c0897b3f689.